File: java/01-beginner/04-control-flow/IfElse.java
/**
 * LESSON: If-Else Statements
 * Control the flow of your program based on conditions.
 */
public class IfElse {
    public static void main(String[] args) {
        // ===== BASIC IF =====
        int age = 18;
        if (age >= 18) {
            System.out.println("You are an adult.");
        }

        // ===== IF-ELSE =====
        int score = 45;
        if (score >= 50) {
            System.out.println("You passed!");
        } else {
            System.out.println("You failed. Score: " + score);
        }

        // ===== IF - ELSE IF - ELSE (Grade System) =====
        int mark = 78;
        System.out.println("\nMark: " + mark);
        if (mark >= 90) {
            System.out.println("Grade: A");
        } else if (mark >= 80) {
            System.out.println("Grade: B");
        } else if (mark >= 70) {
            System.out.println("Grade: C");
        } else if (mark >= 60) {
            System.out.println("Grade: D");
        } else {
            System.out.println("Grade: F");
        }

        // ===== NESTED IF =====
        boolean hasLicense = true;
        int driverAge = 20;
        System.out.println("\n--- Nested If ---");
        if (driverAge >= 18) {
            if (hasLicense) {
                System.out.println("You can drive!");
            } else {
                System.out.println("You need a license first.");
            }
        } else {
            System.out.println("Too young to drive.");
        }

        // ===== TERNARY OPERATOR =====
        // Shorthand: condition ? valueIfTrue : valueIfFalse
        int num = 15;
        String result = (num % 2 == 0) ? "Even" : "Odd";
        System.out.println("\n" + num + " is " + result);

        // Ternary to find max
        int a = 25, b = 30;
        int max = (a > b) ? a : b;
        System.out.println("Max of " + a + " and " + b + " is " + max);
    }
}

File: cpp-practice/01-beginner/04-control-flow/if_else.cpp
/**
 * LESSON: If-Else Statements
 */
#include <iostream>

int main() {
    // ===== BASIC IF =====
    int age = 18;
    if (age >= 18) {
        std::cout << "You are an adult.\n";
    }

    // ===== IF-ELSE =====
    int score = 45;
    if (score >= 50) {
        std::cout << "You passed!\n";
    } else {
        std::cout << "You failed. Score: " << score << "\n";
    }

    // ===== IF - ELSE IF - ELSE =====
    int mark = 78;
    std::cout << "\nMark: " << mark << "\n";
    if (mark >= 90)      std::cout << "Grade: A\n";
    else if (mark >= 80) std::cout << "Grade: B\n";
    else if (mark >= 70) std::cout << "Grade: C\n";
    else if (mark >= 60) std::cout << "Grade: D\n";
    else                 std::cout << "Grade: F\n";

    // ===== NESTED IF =====
    bool hasLicense = true;
    int driverAge = 20;
    std::cout << "\n--- Nested If ---\n";
    if (driverAge >= 18) {
        if (hasLicense) {
            std::cout << "You can drive!\n";
        } else {
            std::cout << "You need a license first.\n";
        }
    } else {
        std::cout << "Too young to drive.\n";
    }

    // ===== TERNARY OPERATOR =====
    int num = 15;
    std::string result = (num % 2 == 0) ? "Even" : "Odd";
    std::cout << "\n" << num << " is " << result << "\n";

    // ===== IF WITH INITIALIZER (C++17) =====
    std::cout << "\n--- If with Initializer (C++17) ---\n";
    if (int val = 42; val > 0) {
        std::cout << val << " is positive\n";
    }
    // 'val' doesn't exist outside the if block

    return 0;
}
